### ISSUE TITLE
Error while running multiagent.py file

### ISSUE THREAD

**Issue by Deepak3994** [madras-simulator/MADRaS#13]
I have done all the preliminary setups for running MADRAS-Simulator. While executing the "multiagent.py" file, I get the following error,

Could not find old network weights
I have been asked to use port:  3001
WARN: gym.spaces.Box autodetected dtype as <class 'numpy.float32'>. Please provide explicit dtype.
WARN: gym.spaces.Box autodetected dtype as <class 'numpy.float32'>. Please provide explicit dtype.
Waiting for server on 3001............
Count Down : 100
Could not find old network weights
I have been asked to use port:  3002
WARN: gym.spaces.Box autodetected dtype as <class 'numpy.float32'>. Please provide explicit dtype.
WARN: gym.spaces.Box autodetected dtype as <class 'numpy.float32'>. Please provide explicit dtype.
Could not find old network weights
I have been asked to use port:  3003
WARN: gym.spaces.Box autodetected dtype as <class 'numpy.float32'>. Please provide explicit dtype.
WARN: gym.spaces.Box autodetected dtype as <class 'numpy.float32'>. Please provide explicit dtype.
Waiting for server on 3001............
Count Down : 99
Waiting for server on 3002............
Count Down : 100
Waiting for server on 3003............
.
.
.
.

May i know where i might have went wrong?
OR
Before executing do we have to start any server?

**Comment by rudrasohan** [comment]
Hi,
The examples and the traffic still use the old API hence, for the time being, are not usable. We will be updating those shortly. For testing the simulator out you can use the latest API and run on baselines, rllab or your own algorithm. For further reference check out [Running the sim](https://github.com/madras-simulator/MADRaS/wiki/Running-Experiments)

**Comment by MehaKaushik** [comment]
Hi @Deepak3994, the master branch is the current developer branch. "Version 1" branch is the most stable current version, please checkout that branch and work with it. Let us know if you face issues there.

**Comment by Deepak3994** [comment]
Hi @MehaKaushik , Sorry i am not able get what is meant by "Version 1". Kindly specify and also provide me the link from where i can checkout stable version of MADRaS. Kindly do the needful.

**Comment by MehaKaushik** [comment]
Try this, @Deepak3994 
https://github.com/madras-simulator/MADRaS/tree/Version1

Terminal commands: 
1. git clone https://github.com/madras-simulator/MADRaS.git
2. cd MADRaS
3. git checkout Version1

You will observe that you will be moved to branch "Version1"

**Comment by Deepak3994** [comment]
Hi @MehaKaushik ,

I have tried the steps which you have mentioned
**Terminal commands:

    git clone https://github.com/madras-simulator/MADRaS.git
    cd MADRaS
    git checkout Version1**

 and have reinstalled torcs, plib and all other which is mentioned in the README.md file.

Steps which i followed:
- Installed torcs-1.3.6 and plib-1.8.5
- Installed the scr-patch 
 **cd scr-patch
    sh do_patch.sh 
    cd ..
    make
    sudo make install
    sudo make datainstall**
- Installed the cpp scr client 
**cd scr-client-cpp
make**
-  Installed the packages from requiments.txt **"sudo -H pip3 install -r requirements.txt"**.

**Requesting you to look into this issue and any suggestions would be appreciated.**

Finally i ran, **"python3 -m example_controllers.behavior_reflex.playGame_DDPG 3001"** under Behavior Reflex. I got the below results which i have shared below,


Output
---------
is_training : 1
Starting best_reward : -10000
600000.0
6000
10000
1
config_file : ~/.torcs/config/raceman/quickrace.xml
2018-11-21 10:54:52.809009: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2018-11-21 10:54:52.884876: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:964] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2018-11-21 10:54:52.885496: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1411] Found device 0 with properties: 
name: Quadro P3000 major: 6 minor: 1 memoryClockRate(GHz): 1.215
pciBusID: 0000:01:00.0
totalMemory: 5.93GiB freeMemory: 5.04GiB
2018-11-21 10:54:52.885508: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1490] Adding visible gpu devices: 0
2018-11-21 10:54:53.089334: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-11-21 10:54:53.089380: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977]      0 
2018-11-21 10:54:53.089387: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990] 0:   N 
2018-11-21 10:54:53.089580: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1103] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 4798 MB memory) -> physical GPU (device: 0, name: Quadro P3000, pci bus id: 0000:01:00.0, compute capability: 6.1)
Could not find old network weights
I have been asked to use port:  3001
WARN: gym.spaces.Box autodetected dtype as <class 'numpy.float32'>. Please provide explicit dtype.
WARN: gym.spaces.Box autodetected dtype as <class 'numpy.float32'>. Please provide explicit dtype.
Waiting for server on 3001............
Count Down : 100
Waiting for server on 3001............
Count Down : 99
Waiting for server on 3001............
Count Down : 98
Waiting for server on 3001............
Count Down : 97
Waiting for server on 3001............
Count Down : 96
Waiting for server on 3001............
Count Down : 95
.
.
.
.

**Comment by MehaKaushik** [comment]
Select scr_server1 in TORCS gui and start race. You are good to go !!!

**Comment by Deepak3994** [comment]
Hi @MehaKaushik ,

The steps which you have given work perfectly good. I have one issue while executing the script that, when the car crashes the torcs stops and it does not relaunch on its own. We keep executing the torcs once it comes out of the gui. 

Can you suggest how to automate the relaunch of the torcs file OR should we manually relaunch everytime when the car crashes? 

Looking forward for your inputs on this issue.

Below is the summary on the torcs that has been terminated,
**Visual Properties Report
------------------------
Compatibility mode, properties unknown.
Waiting for request on port 3001
OpenAL backend info:
  Vendor: OpenAL Community
  Renderer: OpenAL Soft
  Version: 1.1 ALSOFT 1.16.0
  Available sources: 256
  Available buffers: 1024 or more
  Dynamic Sources: requested: 235, created: 235
  #static sources: 21
  #dyn sources   : 235
Terminated**

**Comment by MehaKaushik** [comment]
Hey @Deepak3994 you have to run this script in a terminal, this script should always be running. 
https://github.com/madras-simulator/MADRaS/blob/master/MADRaS/scripts/startTorcs.sh

In line number 9, instead of using "torcs" you can even give the path of torcs directory.

**Comment by bardas** [comment]
I tried the script but The game does not restart.

**Comment by jerrybaba2** [comment]
Hi, dear MADRaS community,
I'm having problems training the MADRaS environment with your example 

"python3 -m example_controllers.behavior_reflex.playGame_DDPG 3001" under Behavior Reflex.
I found this page after I have experienced some problems with it and after reading and following every guide given on this page I have been able to start the training. However, the car is just at a standstill, with no movement even though the process is ongoing. 
To start the training this is what I did:
1. I first run sh startTorcs.sh in MADRaS/scripts in a terminal
2.  I navigate to race>quick race>configure race. then I select Track: Road Tracks: Forza then accept which takes me to select Drivers. 
I selected scr_server 1
then car: car1-trb1, 
category: trb1 
then accept which takes me to Quick race options, 
then we have Race Distance(km)--- I put nothing here.
Laps: 3 then accept.
3. I open a different terminal and run python3 -m example_controllers.behavior_reflex.playGame_DDPG 3001
And attached is what I see, with the car not moving an inch.

![Screenshot from 2021-02-21 17-10-47](https://user-images.githubusercontent.com/50444775/108627661-d5a8e000-7467-11eb-9d3b-86bb34774667.png)

I'm running on ubuntu 20.04 with torcs 1.3.6 installed by the procedure given on the stable version 1 page [here](https://github.com/madras-simulator/MADRaS/tree/Version1)

I would be grateful if someone

 could help me out, how do I make the car move?
Thanks in advance.
